From c56a00fe173a7dd5a8326431ae28863ce432bbc1 Mon Sep 17 00:00:00 2001
From: Paul Moore <paul@paul-moore.com>
Date: Sat, 17 Apr 2021 16:30:48 -0400
Subject: [PATCH] syscalls: update the syscall table to v5.12-rc7

Due to additional ABIs in main we can't do a simple backport or copy
of the syscall table so we are generating it directly in the
release-2.5 branch.

This patch also fixes the missing faccessat2() #defines in the
seccomp-syscalls.h header file.

Signed-off-by: Paul Moore <paul@paul-moore.com>
---
 include/seccomp-syscalls.h | 2 ++
 src/syscalls.csv           | 6 +++++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/include/seccomp-syscalls.h b/include/seccomp-syscalls.h
index 2a4ebd3d..7b69214c 100644
--- a/include/seccomp-syscalls.h
+++ b/include/seccomp-syscalls.h
@@ -564,6 +564,8 @@
 
 #define __SNR_faccessat			__NR_faccessat
 
+#define __SNR_faccessat2		__NR_faccessat2
+
 #ifdef __NR_fadvise64
 #define __SNR_fadvise64			__NR_fadvise64
 #else
diff --git a/src/syscalls.csv b/src/syscalls.csv
index 11d087a6..4c828697 100644
--- a/src/syscalls.csv
+++ b/src/syscalls.csv
@@ -1,4 +1,4 @@
-#syscall (v5.8.0-rc5 2020-07-14),x86,x86_64,x32,arm,aarch64,mips,mips64,mips64n32,parisc,parisc64,ppc,ppc64,riscv64,s390,s390x
+#syscall (v5.12.0-rc7 2021-04-17),x86,x86_64,x32,arm,aarch64,mips,mips64,mips64n32,parisc,parisc64,ppc,ppc64,riscv64,s390,s390x
 accept,PNR,43,43,285,202,168,42,42,35,35,330,330,202,PNR,PNR
 accept4,364,288,288,366,242,334,293,297,320,320,344,344,242,364,364
 access,33,21,21,33,PNR,33,20,20,33,33,33,33,PNR,33,33
@@ -38,6 +38,7 @@ clock_settime64,404,PNR,PNR,404,PNR,404,PNR,404,404,PNR,404,PNR,PNR,404,PNR
 clone,120,56,56,120,220,120,55,55,120,120,120,120,220,120,120
 clone3,435,435,435,435,435,435,435,435,435,435,435,435,435,435,435
 close,6,3,3,6,57,6,3,3,6,6,6,6,57,6,6
+close_range,436,436,436,436,436,436,436,436,436,436,436,436,436,436,436
 connect,362,42,42,283,203,170,41,41,31,31,328,328,203,362,362
 copy_file_range,377,326,326,391,285,360,320,324,346,346,379,379,285,375,375
 creat,8,85,85,8,PNR,8,83,83,8,8,8,8,PNR,8,8
@@ -51,6 +52,7 @@ epoll_create1,329,291,291,357,20,326,285,289,311,311,315,315,20,327,327
 epoll_ctl,255,233,233,251,21,249,208,208,225,225,237,237,21,250,250
 epoll_ctl_old,PNR,214,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR
 epoll_pwait,319,281,281,346,22,313,272,276,297,297,303,303,22,312,312
+epoll_pwait2,441,441,441,441,441,441,441,441,441,441,441,441,441,441,441
 epoll_wait,256,232,232,252,PNR,250,209,209,226,226,238,238,PNR,251,251
 epoll_wait_old,PNR,215,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR,PNR
 eventfd,323,284,284,351,PNR,319,278,282,304,304,307,307,PNR,318,318
@@ -199,6 +201,7 @@ mmap,90,9,9,PNR,222,90,9,9,90,90,90,90,222,90,90
 mmap2,192,PNR,PNR,192,PNR,210,PNR,PNR,89,89,192,PNR,PNR,192,PNR
 modify_ldt,123,154,154,PNR,PNR,123,PNR,PNR,PNR,PNR,123,123,PNR,PNR,PNR
 mount,21,165,165,21,40,21,160,160,21,21,21,21,40,21,21
+mount_setattr,442,442,442,442,442,442,442,442,442,442,442,442,442,442,442
 move_mount,429,429,429,429,429,429,429,429,429,429,429,429,429,429,429
 move_pages,317,279,533,344,239,308,267,271,295,295,301,301,239,310,310
 mprotect,125,10,10,125,226,125,10,10,125,125,125,125,226,125,125
@@ -260,6 +263,7 @@ pread64,180,17,17,180,67,200,16,16,108,108,179,179,67,180,180
 preadv,333,295,534,361,69,330,289,293,315,315,320,320,69,328,328
 preadv2,378,327,546,392,286,361,321,325,347,347,380,380,286,376,376
 prlimit64,340,302,302,369,261,338,297,302,321,321,325,325,261,334,334
+process_madvise,440,440,440,440,440,440,440,440,440,440,440,440,440,440,440
 process_vm_readv,347,310,539,376,270,345,304,309,330,330,351,351,270,340,340
 process_vm_writev,348,311,540,377,271,346,305,310,331,331,352,352,271,341,341
 prof,44,PNR,PNR,PNR,PNR,44,PNR,PNR,PNR,PNR,44,44,PNR,PNR,PNR
